### PR TITLE
polygon/sync: refactor Service.Run with errgroup

### PR DESCRIPTION
**Pull request by battlmonstr**

### CODE REVIEW

File: polygon/heimdall/service.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+		}
+	})
+}
+
+func (s *service) Run(ctx context.Context) error {
+	defer s.checkpointStore.Close()
+	defer s.milestoneStore.Close()
+	defer s.spanStore.Close()
+
+	prepareStoresGroup, _ := errgroup.WithContext(ctx)
```
Comment (AskAlexSharov) [inline polygon/heimdall/service.go:148]: lost `ctx` param. 
child `ctx` will cancel when any goroutine return error/panic

Comment (battlmonstr) [reply polygon/heimdall/service.go:148]: :+1: good spot. Initially the Prepare method didn't accept a ctx, and I've passed a wrong context after a refactoring.

Comment (battlmonstr) [reply polygon/heimdall/service.go:148]: adressed here: https://github.com/ledgerwatch/erigon/pull/10337

File: polygon/heimdall/service.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+	return libcommon.SliceMap(entities, castEntityToWaypoint), err
+}
+
+func (s *service) FetchMilestonesFromBlock(ctx context.Context, startBlock uint64) (Waypoints, error) {
+	s.scraper.Synchronize(ctx)
+	entities, err := s.milestoneStore.RangeFromBlockNum(ctx, startBlock)
+	return libcommon.SliceMap(entities, castEntityToWaypoint), err
+}
+
+func sliceTakeLast[T any](s []T, count int) []T {
```
Comment (taratorio) [inline polygon/heimdall/service.go:115]: this feels like an erigon-lib/common function to me

**Review by taratorio — APPROVED** [review]

**Pull request merged**